ASTree.cpp

```diff
@@ -3290,6 +3290,33 @@ void print_src(PycRef<ASTNode> node, PycModule* mod, std::ostream& pyc_output)
             //pyc_output << ")";
         }
         break;
+    case ASTNode::NODE_CLASS:
+        {
+            pyc_output << "\n";
+            start_line(cur_indent, pyc_output);
+            pyc_output << "class ";
+
+            PycRef<ASTTuple> bases = node.cast<ASTClass>()->bases().cast<ASTTuple>();
+            if (bases->values().size() > 0) {
+                pyc_output << "(";
+                bool first = true;
+                for (const auto& val : bases->values()) {
+                    if (!first)
+                        pyc_output << ", ";
+                    print_src(val, mod, pyc_output);
+                    first = false;
+                }
+                pyc_output << "):\n";
+            }
+            else {
+                pyc_output << ":\n";
+            }
+            printClassDocstring = true;
+            PycRef<ASTNode> code = node.cast<ASTClass>()->code().cast<ASTCall>()
+                ->func().cast<ASTFunction>()->code();
+            print_src(code, mod, pyc_output);
+        }
+        break;
```

Comment:
This looks like a duplicate of the code in `NODE_STORE` (starting on line 3146)...  It would be good to also replace that block with a call to `print_src()` to avoid code duplication.

Reply:
Well, I wish there would be just 1 piece of code for both cases, but after some debugging it turned out that those are different a bit..
eg: in original case with NODE_STORE there are 2 objects, one is ASTName that keeps the class name and other is the code itself
in this case the ASTName is absent, so I"m deducing the class name from the ASTClass object
Just added new commit for that missed printout of class name itself

2) I tried to create a test case, but to be honest I'm not sure how to create a sample one... when I try to define a class at root level its .pyc jumps  into old NODE_STORE case!
thus I need some help here please